### PR TITLE
Simple Schema type support -> pandas dataframe

### PR DESCRIPTION
**Pull request by kumare3**
Adds a new pandas dataframe transformer. This results in a generic schema

### CODE REVIEW

File: flytekit/annotated/workflow.py [hunk 1]
```diff
@@ -152,9 +152,12 @@ def _local_execute(self, ctx: FlyteContext, **kwargs) -> Union[Tuple[Promise], P
...
 
-        if len(output_names) != len(function_outputs):
-            # Length check, clean up exception
-            raise Exception(f"Length difference {len(output_names)} {len(function_outputs)}")
+        if len(output_names) > 1 and isinstance(function_outputs, tuple):
+            if len(output_names) != len(function_outputs):
+                # Length check, clean up exception
+                raise Exception(f"Length difference {len(output_names)} {len(function_outputs)}")
+        elif len(output_names) == 1 and isinstance(function_outputs, Promise):
+            return function_outputs
```
Comment (wild-endeavor) [inline flytekit/annotated/workflow.py:160]: is the stuff below not needed?

Comment (kumare3) [reply flytekit/annotated/workflow.py:160]: what do you mean? its like elif right? the indentation here looks weird, but seems fine in the code

Comment (wild-endeavor) [reply flytekit/annotated/workflow.py:160]: But it returns here right, so the stuff below won't run.  Should it not run?  I added some Promise renaming stuff but maybe it's not needed?

Comment (kumare3) [reply flytekit/annotated/workflow.py]: done

**Pull request merged**